### ISSUE TITLE
JAMstack with Headless CMS is not supported

### ISSUE THREAD

**Issue by k-miyake** [Azure/static-web-apps#38]
In the JAMstack build workflow, it is not possible to get dynamic content from an external API and generate a static site.

Specifically, if I set the `repository_dispatch` as the trigger for GitHub Actions, `Azure/static-web-apps-deploy@v0.0.1-preview` will fail as shown in the screenshot. It's probably a JSON payload issue.

![image](https://user-images.githubusercontent.com/1104403/83330496-3d137400-a2ca-11ea-813a-fcce2492eb09.png)

A typical scenario for JAMstack is to access an online Headless CMS like Contentful in the build workflow to generate static content.

Currently, Azure Static Web Apps doesn't exactly support JAMstack; GitHub Push or PR-based content updates are called SSG (Static Site Generate) rather than JAMstack.

**Comment by aaronpowell** [comment]
Are you able to provide a GitHub repo as an example?

**Comment by k-miyake** [comment]
The repo that have validated this Issue is here,
- https://github.com/k-miyake/vacd-jamstack

During Nuxt.js generate, it gets dynamic contents from Contentful API and generates static files.
As you can see in the workflow history below, it succeeds on the commit trigger, but fails on the repository dispatch(webhook) trigger.
- https://github.com/k-miyake/vacd-jamstack/actions

**Comment by aaronpowell** [comment]
Thank @k-miyake

**Comment by miwebst** [comment]
@k-miyake I'll take a look at this, thanks for reporting it!

**Comment by justinyoo** [comment]
I have the same issue. When I trigger the GitHub Action workflow with the `repository_dispatch` event, it gets the error like the screenshot above. Both `push` and `pull_request` events are fine.

![image](https://user-images.githubusercontent.com/1538528/83909841-79e8da80-a7a4-11ea-8f6b-842ba46123a0.png)

**Comment by miwebst** [comment]
Yes this is a bug. The Build and Deploy Github Action looks for some event data provided by Github, but I believe the repository dispatch event does not provide this. We are working on removing this dependency but it will take some time. For now we can make this error clearer until we have the proper support in place.

**Comment by k-miyake** [comment]
@miwebst Hopefully this will be fixed as soon as possible. If it's going to take a long time, it's better to note it in the documentation FAQ.

**Comment by justinyoo** [comment]
@miwebst Thank you!

Many use cases rely on the `repository_dispatch` event such as manual deployment, scheduled deployment that is not covered by the `schedule` event, or approval process using the other approach - ChatOps or GitOps.

It would be great if this bug gets fixed sooner rather than later.

**Comment by wisdeom** [comment]
I'm also having the same problem with this. Would much appreciate for the fix.

**Comment by miwebst** [comment]
This should be fixed now, please let us know if you continue to run into this issue!

**Comment by k-miyake** [comment]
@miwebst
The `repository_dispatch` trigger makes it possible for me to build and deploy successfully has been checked. 
Here's the GitHub Actions log.
https://github.com/k-miyake/vacd-jamstack/runs/775400365?check_suite_focus=true

Thanks for the quick fix. Great job!
